includes/class-scd-ext-access-control.php

```diff
@@ -170,17 +170,21 @@ public function is_absolute_drip_type_content_blocked( $lesson_id ) {
 		$current_user = wp_get_current_user();
 		$user_id      = $current_user->ID;
 
-		// Convert string dates to date ojbect
+		// Convert string dates to date object.
 		$lesson_drip_date = $this->get_lesson_drip_date( $lesson_id , $user_id );
-		$today            = new DateTime();
+		$today            = Sensei_Content_Drip()->utils->current_datetime()->setTime( 0, 0, 0 );
+
+		if ( ! $lesson_drip_date ) {
+			return $access_blocked;
+		}
 
 		/**
 		 * Compare dates
 		 *
 		 * If lesson drip date is greater than the today
 		 * the drip date ist still active and lesson content should be hidden
 		 */
-		if ( $lesson_drip_date > $today ) {
+		if ( $lesson_drip_date->getTimestamp() > $today->getTimestamp() ) {
```

Comment:
You can compare `DateTime` objects, but I thought sticking to `getTimestamp` would be most explicit.